### ISSUE TITLE
Queries "link" don't work

### ISSUE THREAD

**Issue by gustavorozolin** [isar/isar#370]
I ran the code:

```
final result = await isar!.students.filter()
        .teacher((q) => q.subjectEqualTo('Math')
        .or()
        .subjectEqualTo('English')
    ).findAll();
    print('result ' + result.length.toString());
```
And got the result: `result 0`.

If I remove link query (teacher filter) I get results and on results have `teacher` with **Math** `subject`.

**Version**
 - Platform: iOS
 - Flutter version: 2.10.5
 - Isar version: 2.5.0
 - Isar Generator version: 2.5.0

**Comment by Jictyvoo** [comment]
Having the same problem with Isar `3.0.0-dev.1`

**Comment by leisim** [comment]
Fixed in `3.0.0-dev.8`
